src/data/fin.lean

```diff
@@ -488,6 +488,36 @@ lemma succ_above_lt_gt (p : fin (n + 1)) (i : fin n) : i.cast_succ < p ∨ p < i
 or.cases_on (succ_above_lt_ge p i)
   (λ h, or.inl h) (λ h, or.inr (lt_of_le_of_lt h (cast_succ_lt_succ i)))
 
+/-- Embedding `i : fin n` into `fin (n + 1)` using a pivot `p` that is greater
+results in a value that is less than `p`. -/
+lemma succ_above_lt_iff (p : fin (n + 1)) (i : fin n) : i.cast_succ < p ↔ p.succ_above i < p :=
+begin
+  refine iff.intro _ _,
+  { intro h,
+    rw succ_above_below _ _ h,
+    exact h },
+  { intro h,
+    cases succ_above_lt_ge p i with H H,
+    { exact H },
+    { rw succ_above_above _ _ H at h,
+      exact lt_trans (cast_succ_lt_succ i) h } }
+end
+
+/-- Embedding `i : fin n` into `fin (n + 1)` using a pivot `p` that is lesser
+results in a value that is greater than `p`. -/
+lemma succ_above_gt_iff (p : fin (n + 1)) (i : fin n) : p ≤ i.cast_succ ↔ p < p.succ_above i :=
```

Comment:
Similarly here, I might have named this `lt_succ_above_iff` (i.e. using the position of `lt` to indicate the change, rather than switching to `gt`, which we mostly dislike), and flipped the `iff`.